pints/_optimisers/_nelder_mead.py

```diff
@@ -0,0 +1,337 @@
+
+        # Check ask/tell
+        if self._ready_for_tell:
+            raise Exception('ask() called twice')
+        self._ready_for_tell = True
+
+        # Initialise
+        if self._xs is None:
+            self._running = True
+
+            # Initialise simplex, reticulate splines
+            # TODO: Use sigma here?
+            n = self._n_parameters
+            self._xs = np.zeros((n + 1, n))
+            self._xs[0] = self._x0
+            x_grow = 1.05       # From scipy implementation
+            x_zero = 0.00025    # From scipy implementation
+            for i in range(n):
+                self._xs[1 + i] = self._x0
+                if self._xs[1 + i][i] == 0:
+                    self._xs[1 + i][i] = x_zero
+                else:
+                    self._xs[1 + i][i] *= x_grow
+
+            # Ask for initial points
+            return np.array(self._xs, copy=True)
+
+        # Shrink operation
+        if self._shrink:
+            for i in range(self._n_parameters):
+                self._xs[1 + i] = \
+                    self._xs[0] + self._ys * (self._xs[1 + i] - self._xs[0])
+
+            return np.array(self._xs[1:], copy=True)
+
+        # Start of normal iteration, ask for reflection point
+        if self._xr is None:
+
+            # Order, and calculate mean of all except worst
+            ifs = np.argsort(self._fs)
+            self._xs = self._xs[ifs]
+            self._fs = self._fs[ifs]
+            self._xm = np.mean(self._xs[:-1], axis=0)
+
+            # Calculate reflection point and return
+            self._xr = self._xm + self._dr * (self._xm - self._xs[-1])
+
+            # Ask for f(xr)
+            return np.array([self._xr])
+
+        # Extended iteration: ask for expansion or contraction point
+        return np.array([self._xp])
+
+    def fbest(self):
+        """ See: :meth:`pints.Optimiser.fbest()`. """
+        if not self._running:
+            return float('inf')
+        return self._fs[0]
+
+    def name(self):
+        """ See: :meth:`pints.Optimiser.name()`. """
+        return 'Nelder-Mead'
+
+    def running(self):
+        """ See: :meth:`pints.Optimiser.running()`. """
+        return self._running
+
+    def stop(self):
+        """ See: :meth:`pints.Optimiser.stop()`. """
+        return False
+
+    def tell(self, fx):
+        """ See: :meth:`pints.Optimiser.tell()`. """
+
+        # Check ask/tell sequence
+        if not self._ready_for_tell:
+            raise Exception('ask() not called before tell()')
+        self._ready_for_tell = False
+
+        # Initialise
+        if self._fs is None:
+            fx = np.array(fx, copy=True)
+            if np.prod(fx.shape) != self._n_parameters + 1:
+                raise ValueError(
+                    'Expecting a vector of length (1 + n_parameters).')
+            self._fs = fx.reshape((1 + self._n_parameters, ))
+
+            # Return: ask will set xr, get f(xr)
+            return
```

Comment:
Due to my Pythonic ignorance, I don't know what return on its own does?

Reply:
Its equivalent to `return None`. I'll change to that explicitly

Reply:
Actually I won't! Got a bit confused there with the MCMC `tell()`, which returns something or doesn't in special cases. Optimiser tell()s never return anything, so will stick to `return` meaning "exit the function now" (and assumes no value needs returning)